### PR TITLE
Fix the chaos while clicking in panel mode

### LINKED ISSUE #2831: Repeated mouse click handling causes chaos in panel mode
# Repeated mouse click handling causes chaos in panel mode

Under the panel mode (`v`), when the user clicks the items in function/symbol lists, or addresses in the disassembly panel, the panels should jump to the corresponding address. However, since the click contains **up** and **down**, the handler processes the click two times. With the scroll reset, repeated handling causes chaos in use.

For example, a function list (`afl` panel) with windows size = 3, as follows:

```
--------- afl panel
addr1
addr2
addr3
---------
addr4
```
when u scroll to:
```
addr1
--------- afl panel
addr2
addr3
addr4
---------
```
then click addr2, the panel actually jumps to addr1.

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

CI is green.

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/2831

### CODE REVIEW

File: librz/core/tui/panels.c [hunk 1]
```diff
@@ -1953,6 +1954,14 @@ static bool __handle_mouse_on_X(RzCore *core, int x, int y) {
 }
 
 static bool __handle_mouse_on_panel(RzCore *core, RzPanel *panel, int x, int y, int *key) {
+	// mouse click contains down and up
+	// thus use on_mouse_down to avoid repeated process
+	static bool on_mouse_down = false;
```
Comment (XVilka) [inline librz/core/tui/panels.c]: We advise against using static variables and global state:
- https://github.com/rizinorg/rizin/issues/1874
- https://github.com/rizinorg/rizin/issues/276

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/tui/panels.c (hunk 1 above)
Comment (ret2libc) [reply librz/core/tui/panels.c]: Is there a way to distinguish between the up and down mouse events?

**Comment by PeiweiHu** [comment]
The original `mouse_event` contains `0` and `1` to indicate whether the mouse event happens. This pr defines an enumeration to represent the specific mouse event type. Especially, the value `MOUSE_DEFAULT` represents an event with no consideration of a specific type to imitate `mouse_event = 1` before changing.

File: librz/cons/cons.c [hunk 2]
```diff
@@ -474,11 +474,11 @@ void resizeWin(void) {
 	}
 }
 
-RZ_API void rz_cons_set_click(int x, int y) {
+RZ_API void rz_cons_set_click(int x, int y, MouseEvent event) {
```
Comment (XVilka) [inline librz/cons/cons.c:483]: Please add a Doxygen.

**Review by XVilka — CHANGES_REQUESTED** [review]
@GustavoLCR please take a look if this logic would work in Windows console[s] as well.

File: librz/include/rz_cons.h [hunk 3]
```diff
@@ -496,6 +496,19 @@ typedef struct rz_cons_context_t {
 
 #define HUD_BUF_SIZE 512
 
+typedef enum {
+	MOUSE_NONE,
```
Comment (ret2libc) [inline librz/include/rz_cons.h]: i think some code still assumes that MOUSE_NONE == 0? if so, do `MOUSE_NONE = 0,` please

**Review by GustavoLCR — APPROVED** [review]

File: librz/include/rz_cons.h (hunk 3 above)
Comment (wargio) [inline librz/include/rz_cons.h:500]:
```suggestion
	MOUSE_NONE = 0,
```

Comment (wargio) [reply librz/include/rz_cons.h]:
```suggestion
	MOUSE_NONE = 0,
```

**Review by XVilka — APPROVED** [review]

**Pull request merged**